### ISSUE TITLE
Code blocks with email-like text are no longer parsed as code

### ISSUE THREAD

**Issue by kornelski** [markedjs/marked#1218]
**Marked version:** 0.3.12 (OK) 0.3.14 (buggy)

```markdown
`user@host`
```

used to render as `<code>` block:

`user@host`

but now it's rendered as an e-mail link with literal `` ` `` in it:

[\`user@host\`](mailto:user@host)

**Comment by UziTech** [comment]
seems to work in the latest 0.3.19: [demo](https://marked.js.org/demo/?outputType=html&text=%60user%40host%60)

**Comment by kornelski** [comment]
Try

```

 `/users?email=joe@smith.com`.

```

**Comment by UziTech** [comment]
Yup ` `user@host.com` ` does it

**Comment by joshbruce** [comment]
Looking at the specs to try and figure out a test to create and they don't seem to call out this specific use case.

For CommonMark:

> Autolinks are absolute URIs and email addresses inside < and >.

So, this would be something related to GitHub Flavored Markdown:

> An extended email autolink will be recognised when an email address is recognised within any text node. 

Looking at code spans (examples 322 to 327) I would suspect that the proper behavior would be something like:

```
Markdown:

`user@example.com`

Html:

<code>user@example.com</code>
```

@kornelski: Is that what you're looking for?

**Comment by joshbruce** [comment]
`someone@example.com`

**Comment by kornelski** [comment]
In my case `<code>` is more important than `<a>`, because I'm using a made-up e-mail address in documentation and don't want it linked. If you prefer links, then both (`<code><a>email</a></code>`) would be a reasonable rendering IMHO.

**Comment by UziTech** [comment]
IMP creating a link without `<>` surrounding it should only happen when the gfm option is true since this would not be compliant with commonmark

**Comment by joshbruce** [comment]
GFM complies with CommonMark; so, autolinking should be invoked by the `<>` combination. Given what I can discern from the code span area of CM and GFM, I would say the code span leaves everything inside alone. The linked PR runs a test that would result in that ignore.

**Comment by UziTech** [comment]
in order to pass [CommonMark Example #583](https://spec.commonmark.org/0.28/#example-583) This autolinking without `<>` will have to be moved to `gfm: true`

**Comment by joshbruce** [comment]
Think this conversation might go beyond this issue...deeper issues with Marked at the moment...making a new issue:

**Comment by joshbruce** [comment]
To pass I think we actually need a way to specify just CommonMark - not GFM or Pedantic. We currently don't have anything like that. The CommonMark Autolinks remain in the CommonMark spec tests - where they are at present - of which 583 is already listed as failing but should pass.

**Comment by Feder1co5oave** [comment]
According to gfm spec, autolinks have [higher precedence](https://github.github.com/gfm/#example-333) than code spans.

Also, email addresses are recognized and autolinked by the `InlineLexer.rules.gfm.url` rule, which (re-)uses the `InlineLexer.rules._email` subrule. This comes from cm [spec](https://spec.commonmark.org/0.28/#email-address). This is wrong, since gfm defines extended autolinked email adresses [quite differently](https://github.github.com/gfm/#extended-email-autolink).

I'm making a PR for this

**Comment by UziTech** [comment]
this is fixed by #1337

**Comment by Feder1co5oave** [comment]
Aw, that PR was not linked :(

I have done a totally different solution, I'm not sure what you were going for with that PR. I'll just submit it and then you'll decide what to pick and/or integrate one with the other!
